packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -237,6 +238,14 @@ export type TableProps<
    * @beta
    */
   enableVirtualization?: boolean;
+  /**
+   * Initialize the table scrolled to this item.
+   *
+   * Must be used in conjunction with `getRowId` prop.
```

Comment:
We might need to add example in here.. cause it is not obvious at all with this :/
Maybe we could have single prop instead, which would be object for item and comparer?
`scrollToItem: {item: T, comparer: row => string}` ?
We could even have default comparer with ids or so.. not sure

Reply:
I spent a very long time trying to get the TypeScript props to require `getRowId` if `scrollToItem` was passed, but it always fell short. There is an example in the code sandbox which shows with the extra needed prop.

I wouldn't want to pass another comparer, as the react-table already has a getRowId prop that can be used. But I'm open to whatever the maintainers want to do.

Reply:
hmm... Have you tried conditional types?
Something like this, maybe...
```ts
| {
      scrollToItem: undefined;
      getRowId?: (originalRow: T, relativeIndex: number, parent?: Row<T>) => string;
  }
| {
      scrollToItem: T;
      getRowId: (originalRow: T, relativeIndex: number, parent?: Row<T>) => string;
  }
```
https://www.benmvp.com/blog/conditional-react-props-typescript/

Reply:
Ah, you [nerd-sniped](https://xkcd.com/356/) me! I still can't get it to play nice on the actual table. Playing around with dummy types, it looks like it should work, but it doesn't play nicely in the actual table code. Maybe you can crack it?

Reply:
This implementation actually does work how I want in my app. If I leave off `scrollToItem`, I get an error, and I can use or not use `getRowId` without any issues.

```ts
type ScrollToItemProps<T extends Record<string, unknown>> =
  | {
      scrollToItem: T;
      getRowId: TableOptions['getRowId'];
    }
  | {
      scrollToItem?: never;
      getRowId?: TableOptions['getRowId'];
    };
```

One annoyance is that the typings in react-table aren't great, and I have to do a work around to prevent a compile error

```ts
getRowId={(row: PWData | object) => (row as PWData).instanceId}
// shouldn't need to use `| object`, but otherwise it gives 
// Type '(row: PWData) => string' is not assignable to type ((originalRow: PWData ...) => string) & ((originalRow: {} ...) => string)'
```

Reply:
Yeah.. I think we have issues in storybook after monorepo change...
I dont see the code pushed, maybe you forgot? Would like to check in Vite app too.

Reply:
Didn't push it, but I will now.

Reply:
hm... How did you verify that it works? It doesnt at my end, I think :/

Reply:
I did a build and moved the esm and cjs files into my app to replace what's in the iTwin package. I have a lot of trouble running the iTwin-react stuff on my computer; it complains about the version of Node I'm using, and when I change it, then my other projects are grumpy. I'm happy not to have this extra complication around the props if we can't get it to play nicely without a bunch of effort.

Reply:
Ok, I'll be able to get to the tests today or tomorrow. I'm having trouble getting them to work nicely with virtualization, though. There is a lot of mocking going on around all the virtualization code in the tests, and I can't follow it all...

Reply:
Let me know how it goes. I could try to have a look at those.